src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -176,20 +168,17 @@ public virtual void CopyTo(Stream destination, int bufferSize)
             StreamHelpers.ValidateCopyToArgs(this, destination, bufferSize);
 
             byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
-            int highwaterMark = 0;
             try
             {
                 int read;
                 while ((read = Read(buffer, 0, buffer.Length)) != 0)
                 {
-                    if (read > highwaterMark) highwaterMark = read;
                     destination.Write(buffer, 0, read);
                 }
             }
             finally
             {
-                Array.Clear(buffer, 0, highwaterMark); // clear only the most we used
```

Comment:
Any reason why you removed this?

Reply:
Noted in the PR description:
"(While I was modifying this, I also removed some unnecessary array clearing that we'd added before later deciding it wasn't needed in general.)"
We'd initially gone through and adding clearing everywhere we were using array pool.  Then we subsequently decided we didn't need to, that we would only explicitly clear in very explicitly high-security code (like when using an array to hold a key in a crypto lib), and we removed some of these and stopped adding more, but we missed some.

Reply:
Ah